### PR TITLE
feat: add parse_null function and update filtering tests for minimum …

### PR DESCRIPTION
**Pull request by atrigila**
…abundance

For: https://github.com/nf-core/differentialabundance/pull/583

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Broadcast software version numbers to `topic: versions` - [See version_topics](https://nf-co.re/blog/2025/version_topics)
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by grst** [comment]
I think the meta.yaml still needs to be updated? Should be fine other than that

**Comment by atrigila** [comment]
Sorry if I am missing something but what would you update from the meta.yml? The inputs have not changed and the new changes are related external arguments (which are generally not documented).

**Comment by grst** [comment]
Sorry, my bad, I somehow thought the external parameters were part of meta.yml.

I still think that it should be documented *somewhere* that `minimum_abundance` can now take `null/false`. In general, the usage of the whole module would be pretty unclear without additional context. Could you add something to the top comment? Or maybe argument descriptions in the `opt <- list( ... )` definition?

File: modules/nf-core/custom/matrixfilter/templates/matrixfilter.R [hunk 1]
```diff
@@ -15,6 +15,15 @@
...
 # between those groups.
+#
+# This script supports the following options:
+#
+# - `--minimum_abundance`: Minimum value threshold for feature filtering.
+#   Set to a numeric value (e.g., 0.5) to enable, or to 'false'/'null' to disable.
+# - `--minimum_samples`: Minimum number of samples passing the abundance threshold.
+# - `--minimum_samples_not_na`: Minimum number of non-NA values per feature.
+# - `--grouping_variable`: Optional column in sample sheet for group-specific filtering.
+# - `--minimum_proportion`: Proportion-based filtering threshold.
```
Comment (atrigila) [inline modules/nf-core/custom/matrixfilter/templates/matrixfilter.R:26]:
```suggestion
# - `--minimum_proportion`: Proportion-based filtering threshold.
# - `--minimum_proportion_not_na`: Minimum proportion of non-NA values required per feature.
# - `--most_variant_features`: Optional integer specifying the number of most-variant features to retain.
```

**Comment by atrigila** [comment]
@grst done
